### ISSUE TITLE
Setting instance_profile_credentials_retries across the board

### ISSUE THREAD

**Issue by jkburges** [aws/aws-sdk-ruby#1774]
Please fill out the sections below to help us address your issue

### Issue description

I am wondering: is it possible to set `instance_profile_credentials_retries` across the board so that *any* AWS client gets the value.

e.g.: 

```ruby
Aws.config.update({
  instance_profile_credentials_retries: 5
})
```

It sort of looks as though that might be the case [in code](https://github.com/aws/aws-sdk-ruby/commit/aa4143487ee0c3b71185afcd6493b2d4d8c961b3#diff-1d69d51d6bd7de6aa6817464c75323baR27) but I don't know the code well enough to be sure.

I know that it can be done for each client, i.e.: 

```ruby
  s3 = Aws::S3::Client.new(
    instance_profile_credentials_timeout: 5
  )
 ```

### Gem name ('aws-sdk', 'aws-sdk-resources' or service gems like 'aws-sdk-s3') and its version

aws-sdk v3

**Comment by cjyclaire** [comment]
@jkburges Yes, it's [possible](https://github.com/aws/aws-sdk-ruby/blob/9ea447b7e9e57e27bb7924aca40a5bf0c3265a94/gems/aws-sdk-core/lib/aws-sdk-core/plugins/credentials_configuration.rb#L56) : )
